### PR TITLE
WIP: add demo

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by pavoljuhas** [comment]
@CJ-Wright - I have put together a draft of ES setup for importing the analysis run_start document that you sent a while ago - see https://github.com/CJ-Wright/sciprovenance/compare/csi_demo...pavoljuhas:add-tomo-search.

This thus far defines a document mapping between the run-start fields and Elasticsearch and has the following output:
```sh
$ python dbes_tomo_example.py
Sample ES document:

{'_id': '38d93d78-33a7-4a83-a8fe-c132e1fc1c45',
 'analysis_stage': 'img_sinogram',
 'puid': ['fe416836-a19f-40fa-8d79-c80d4a3dbc65'],
 'uid': '38d93d78-33a7-4a83-a8fe-c132e1fc1c45',
 'usednodes': [{'ndfunc': 'img_sinogram',
                'ndtype': 'shed.translation.ToEventStream'},
               {'ndfunc': 'xpdtools.pipelines.tomo.conditional_squeeze',
                'ndtype': 'rapidz.core.map'},
               {'ndfunc': 'xpdtools.pipelines.tomo.sort_sinogram',
                'ndtype': 'rapidz.core.starmap'},
               {'ndfunc': 'xpdtools.pipelines.tomo.append_data',
                'ndtype': 'rapidz.core.accumulate'},
               {'ndtype': 'rapidz.core.zip'},
               {'ndfunc': 'tomopy.prep.normalize.minus_log',
                'ndtype': 'rapidz.core.map'},
               {'ndfunc': 'xpdtools.pipelines.tomo.reshape',
                'ndtype': 'rapidz.core.map'},
               {'ndfunc': 'data', 'ndtype': 'shed.translation.FromEventStream'},
               {'ndfunc': 'numpy.deg2rad', 'ndtype': 'rapidz.core.map'},
               {'ndfunc': 'data',
                'ndtype': 'shed.translation.FromEventStream'}]}
```

Would this work as an ES summary of the tomography pipeline?

File: demo/data_an.ipynb [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# query an_es/databroker for tomo recon\n",
+    "hdrs = an_db_es()\n",
```
Comment (CJ-Wright) [inline demo/data_an.ipynb]: @pavoljuhas Can you put in a valid search query here?

Comment (pavoljuhas) [reply demo/data_an.ipynb]: sure - here are same examples

```patch
diff --git a/demo/data_an.ipynb b/demo/data_an.ipynb
index ea2a439..3456c17 100644
--- a/demo/data_an.ipynb
+++ b/demo/data_an.ipynb
@@ -62,7 +62,13 @@
    "outputs": [],
    "source": [
     "# query raw es for dino tooth, print output\n",
-    "raw_es.qsearch()"
+    "raw_es.qsearch()\n",
+    "vqraw = lambda q: print(q, raw_es.qsearch(q))\n",
+    "vqraw('dinosaur')  # search all fields\n",
+    "vqraw('sample_name:dinosaur')  # search specific field\n",
+    "vqraw('dino*')  # glob-like search\n",
+    "vqraw('dinosaurus~2')  # fuzzy search max edit distance of 2\n",
+    "vqraw('/saur/')  # regular expression search"
    ]
   },
   {
@@ -71,9 +77,18 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# test queries with an_es\n",
+    "vqan = lambda q: print(q, an_es.qsearch(q))\n",
+    "vqan('img_sinogram')\n",
+    "# assuming 'puid': ['fe416836-a19f-40fa-8d79-c80d4a3dbc65']\n",
+    "vqan('puid:fe416836')  # word in puid\n",
+    "vqan('puid:/a19f/')  # re-search items in puid\n",
+    "vqan('analysis_stage:img_sinogram')\n",
+    "vqan('usednodes.ndfunc:*sort_sinogram')         \n",
+    "\n",
     "# query an_es/databroker for tomo recon\n",
     "hdrs = an_db_es()\n",
-    "tomo_analysis_hdr = next(iter(hdrs))\n",
+    "tomo_analysis_hdr = next(hdrs)\n",
     "print(tomo_analysis_hdr.stop)"
    ]
   },
```

Comment (pavoljuhas) [reply demo/data_an.ipynb]:
```
git pull https://github.com/pavoljuhas/sciprovenance.git x-pr10-esqueries
```

Comment (CJ-Wright) [reply demo/data_an.ipynb]: (you could push to my branch btw)

**Comment by pavoljuhas** [comment]
For the demo let us disable warnings in notebook - for example by setting
`export PYTHONWARNINGS=ignore` in parent shell.

**Comment by pavoljuhas** [comment]
@CJ-Wright - please pull to export kwargs to ES
```
git pull https://github.com/pavoljuhas/sciprovenance add-ndkwargs-field
```

**Comment by pavoljuhas** [comment]
@CJ-Wright - are you OK with merging this in?
cc: @hjjvandam

**Pull request merged**